.circleci/config.yml

```diff
@@ -0,0 +1,52 @@
+version: 2
+jobs:
+  build:
+    docker:
+      - image: linkerd/rustup-nightly:v1
+
+    working_directory: ~/tacho
+    steps:
+      - run:
+          name: update rust nightly
+          command: rustup update nightly
+
+      - restore_cache:
+          key: rust-{{ checksum "/rust/update-hashes/nightly-x86_64-unknown-linux-gnu" }}.0
+
+      - run:
+          name: install rustfmt and clippy
+          command: |
+            if [ ! -x /cargo/bin/cargo-fmt ]; then cargo install rustfmt ; fi
+            if [ ! -x /cargo/bin/cargo-clippy ]; then cargo install clippy ; fi
+
+      - save_cache:
+          key: rust-{{ checksum "/rust/update-hashes/nightly-x86_64-unknown-linux-gnu" }}.0
+          paths:
+            - /rust
+            - /cargo
```

Comment:
just confirming: the checksum is on a file in `/rust/` but you're restoring `/cargo`. is it ok if the `cargo install ...` commands would have installed newer versions of themselves, but did not due to caching?